samza-core/src/main/scala/org/apache/samza/checkpoint/OffsetManager.scala

```diff
@@ -284,6 +319,14 @@ class OffsetManager(
     } else {
       debug("Skipping checkpoint manager shutdown because no checkpoint manager is defined.")
     }
+
+    if (startpointManager != null) {
```

Comment:
I think it would be better to move the startpoint manager outside of offsetmanager. The utility of the startpointmanager ends after the container startup sequence. But the offsetmanager lifecycle is tied to the container lifecycle. It would be better to stop the startpoint manager after we update the offset and delete the startpoint and not have the idle metadata-store connection lurking around.